src/gui/search/searchwidget.cpp

```diff
@@ -242,12 +241,18 @@ QTabWidget *SearchWidget::searchTabs() const
 void SearchWidget::on_searchButton_clicked()
 {
     if (Utils::Misc::pythonVersion() < 0) {
-        m_mainWindow->showNotificationBaloon(tr("Search Engine"), tr("Please install Python to use the Search Engine."));
+        // unconditional notifications, user can not disable it
+        Notifications::Request()
+            .title(tr("Search Engine"))
+            .message(tr("Please install Python to use the Search Engine."))
+//          .setWidget(this)
+            .severity(Notifications::Severity::Error)
+            .exec();
```

Comment:
>The invocation of the notification here is direct because this notification has to be shown unconditionally

We have both conditionally and unconditionally showed notifications in Subject Domain so model must reflect this circumstance but not to hide it by using some backdoors and breaking selected design pattern.
>I did not want to complicate that code by adding an option to suppress disabling the notification.

Your code is complicated now. But you don't want to see this, considering it excellent.
If your model will fit the subject domain and you won't "break" the use of a single pattern in its implementation you will get rid of many unnecessary things and dependencies.